python/pyspark/ml/wrapper.py

```diff
@@ -190,8 +190,30 @@ def _transform(self, dataset):
         return DataFrame(self._java_obj.transform(dataset._jdf), dataset.sql_ctx)
 
 
+class JavaCallable(object):
```

Comment:
`JavaWrapper` does define a java object wrapper variable, but also does more since it is derived from `Params`.  Since I need to wrap a Java Summary object (which is just a plain Java object), it doesn't really make sense to use it unless we want to change that.

Actually, I think it might make more sense to have `JavaWrapper` inherit from `JavaCallable` and only define `_java_obj` there.  Then rename the classes a little like this to better reflect what they are for:

`JavaCallable(object)` -> `JavaWrapper(object)`
and
`JavaWrapper(Params)` -> `JavaParamWrapper(JavaWrapper, Params)`

That might be better done in another JIRA though, because of the name change